### ISSUE TITLE
Bootstrap debops via Ansible provisioner (e.g. Vagrant)

### ISSUE THREAD

**Issue by ganto** [debops/ansible-debops#7]
After my first attempt went to the wrong place ([ansible-bootstrap#20](https://github.com/debops/ansible-bootstrap/pull/20)), I'll continue discussing my issue here.

My current idea is to extend DebOps so that it can be bootstrapped and triggered via external Ansible provisioner. Ideally this includes:
* Setting up Ansible and DebOps on a fresh Linux host (VM image or container)
* Clone or generate a DebOps project
* Trigger a DebOps run

This should make it possible to distribute a fully featured DebOps environment via a Vagrant project.

Currently in my minimal `Vagrantfile` I have something like this:

      config.vm.define :master do |master|
        [...]
        config.vm.provision "ansible" do |ansible|
          ansible.host_key_checking = "false"
          ansible.playbook          = "bootstrap-debops.yml"
          ansible.groups            = {
            "debops_master"         => ["master"],
            "debops_master:vars"    => {
              "debops_ansible_packages"   => "[]",
              "debops_install_systemwide" => "False",
            },
          }

The `bootstrap-debops.yml` looks like this:

```
- hosts: debops_master
  gather_facts: false
  become: true
  become_user: root

  pre_tasks:

    - name: Download bootstrap-ansible.sh script
      get_url:
        url: 'https://raw.githubusercontent.com/debops/debops/master/misc/scripts/bootstrap-ansible.sh'
        dest: '/usr/local/sbin'
        mode: '0755'

    - name: Build Ansible Debian package
      command: '/usr/local/sbin/bootstrap-ansible.sh'
      args:
        creates: /usr/bin/ansible

  roles:

    - role: ansible-debops
```
 
Of course there is some more magic involved like cloning the `ansible-debops` repository prior to executing the Vagrant provisioning via Ansible, but this shouldn't matter atm.

If I run this, I have Ansible and DebOps more or less setup. As you can see, the variables for the `ansible-debops` role can be defined in the `Vagrantfile`. Still there are some irritating and missing things where I need some advice:
* I set `debops_install_systemwide: False` in the hope I would end up with a local installation as described in the [DebOps Getting Started](http://docs.debops.org/en/latest/debops/docs/getting-started.html). However, I end up without any DebOps playbooks and roles. Is this on purpose or is there still something missing? I would like to have the roles available as regular user, so that I could login via `vagrant ssh` and have the familiar environment as if I setup DebOps manually with my user.
* After setting up the roles and playbooks I might want to run `debops` in a later task. Is there anything which would speak against a `debops_update_method: sync`?
* What I'm still missing is a DebOps project. Would it be accepted if I port the [bootstrap_project_...](https://github.com/debops/ansible-bootstrap/pull/20/files#diff-4d2f812737977bf2aff5b6185d5a01f6R14) parts to `ansible-debops`?

I'm fine to send patches for all the issues mentioned here. I just like to ask before heading to the wrong direction (again).

**Comment by drybjed** [comment]
I suppose that adding support for user-level roles and playbooks shouldn't be that difficult. A synchronyous role installation can also be added as an option. Since role has small amount of tasks at the moment, we can experiment. :-)

Cloning a specified repository or project initialization as a task sounds good as well. Do you think that adding an option do manage DebOps on a different account than the administrator (well, `vagrant` in this case) could be useful? In that case doing the tasks with `become_user: '{{ debops_user }}' sounds best.

If you want to mess with the role, you could also rename the `debops_*` variables to `debops__*` to better indicate role namespace in the future.

**Comment by ganto** [comment]
With the recently merged PRs (#8, #9, #10, #11) I'm finally able to easily bootstrap and run DebOps with a minimal Ansible playbook. See [examples#12](https://github.com/debops/examples/pull/12).
